include/sys/types.h

```diff
@@ -175,6 +183,16 @@ typedef int wint_t;
 
 typedef int wctype_t;
 
+#if defined(CONFIG_FS_LARGEFILE) && defined(CONFIG_HAVE_LONG_LONG)
+/* Large file versions */
+
+typedef uint64_t     fsblkcnt_t;
+typedef uint64_t     fsfilcnt_t;
+
+typedef uint64_t     blkcnt_t;
+typedef int64_t      off_t;
```

Comment:
block counts and file counts don't have anything to do with large file support, do they?  The are related to volume size, but that is a completely different thing from file size.

Reply:
fsfilcnt_t may not, but block count is related to large file support. For example, my laptop buyed two years ago has a SSD with 2TB, if the sector is 512B, 32bit block count can't represent my old SSD correctly. Other libc has the similar change:
https://github.com/bminor/musl/blob/cfdfd5ea3ce14c6abf7fb22a531f3d99518b5a1b/include/sys/types.h#L74-L80
https://github.com/bminor/glibc/blob/595c22ecd8e87a27fd19270ed30fdbae9ad25426/posix/bits/types.h#L179-L189